TypeCobol/Compiler/CodeElements/Expressions/Expression.cs

```diff
@@ -14,6 +17,19 @@ public abstract class Expression : IVisitable
 
 		public ExpressionNodeType NodeType { get; private set; }
 
+        /// <summary>
+        /// Returns the characteristic element of a node
+        /// For a operation or relation, this would be the operator
+        /// For a leaf, this would be the name of the variable, ...
+        /// </summary>
+        public abstract object NodeData();
```

Comment:
Use a property instead of a method, we should assume that retrieving the `NodeData` is not a costly operation.

Reply:
So NodeData is not good name, here it's a CodeElement, this seems to be a Label (étiquette in french)

Reply:
This is neither a Node nor a CodeElement, the class Expression inherits from IVisitable only. The name `Label` stays relevant nonetheless.

The true purpose of this method is to return an object that allows to say if the Labels of 2 Expressions are equal. Do you think the property Label has a value by itself or would a method `HasSameLabel(Expression other)` be better?

Reply:
I still thiking that HasSameLabel is better because it's for testing Expression Kinds and for a lead named expression value.

Reply:
`Label` is interesting, it would be typed as `string` and be easier to use from client perspective. I'm still looking for a name that conveys the idea that this "label" is linked to the nature of the expression...

Reply:
What about "Discriminator" or "DiscriminatorLabel" ? The word "discriminator" is used in Hibernate persistence framework for a string property that allows to distinguish between types among a class hierarchy.